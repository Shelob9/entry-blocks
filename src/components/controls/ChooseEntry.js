import React, {Fragment} from 'react';

export const ChooseEntry = (props) =>{
	const id = 'caldera-forms-entry-chooser-' + props.instanceId;
	let options = [
		{
			label:'--',
			value: null
		}
	];

	if( JSON.stringify(props.entries)!== JSON.stringify({})){
		Object.values(props.entries).forEach( (entry) => {
			if( 'object' === typeof  entry ){
				options.push( {
					value:entry.id,
					label:entry.id
				});
			}

		});
	}
	return <Fragment>
		<label
			htmlFor={id}
		>
			Choose Entry
		</label>
		<select
			id={ id }
			className={'caldera-forms-entry-chooser'}
			value={ props.currentEntry }
			onChange={ (event) => {props.onSetEntry( event.target.value)} }
		>
			{ options.map( (option) => ( <option key={option.value} value={option.value}>{option.label}</option>) )}
		</select>
	</Fragment>
};
