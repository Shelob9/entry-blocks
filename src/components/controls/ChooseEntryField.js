
import classNames from "classnames";
import {Fragment} from '@wordpress/element';

export function getFormFieldsOfForm(form) {
	return  {
		...form.field_details.entry_list,
		...form.field_details.order
	};
}

export const ChooseEntryField = (props) => {
	const {
		hideLabel,
		currentEntry,
		instanceId,
		form,
		onSetField,
		entryFieldId
	} = props;
	const formFields = getFormFieldsOfForm(form);
	const className = 'caldera-forms-entry-field-chooser';
	const id = className + '-' + instanceId;

	let options = [
		{
			label: '--',
			value: null
		}
	];


	if ('object' === typeof formFields && JSON.stringify(formFields) !== JSON.stringify({})) {
		Object.values(formFields).forEach((field) => {
			if ('object' === typeof  field) {
				options.push({
					value: field.id,
					label: field.label
				});
			}

		});
	}
	return (
		<Fragment>
			<label
				htmlFor={id}
				className={classNames({
					'sr-only': hideLabel,
					'screen-reader-text': hideLabel,
				})}
			>
				Choose Field
			</label>
			<select
				id={id}
				className={className}
				value={entryFieldId}
				onChange={(event) => {
					onSetField(event.target.value)
				}}
			>
				{options.map((option) => (
					<option key={option.value} value={option.value}>{option.label}</option>))}
			</select>
		</Fragment>

	);
};
