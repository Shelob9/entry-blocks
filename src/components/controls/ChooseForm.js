import React from 'react';
import classNames from 'classnames';
export const ChooseForm = (props) => {
	const id = 'caldera-forms-form-chooser-' + props.instanceId;
	return (
		<div>
			<label
				className={classNames({'sr-only': false})}
				htmlFor={id}
			>
				Choose Form
			</label>
			<select
				key={id}
				id={id}
				className={'caldera-forms-form-chooser'}
				value={ props.currentFormId }
				onChange={ (event) => {props.onSetForm(event.target.value)} }
			>
				{<option key={'null-opt'}> -- </option>}
				{ props.forms.map( form => ( <option key={form.ID} value={form.ID}>{form.name}</option>) )}
			</select>
		</div>
	);

};
