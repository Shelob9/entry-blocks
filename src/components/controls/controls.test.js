import renderer from 'react-test-renderer';
import React from 'react';
import {ChooseForm} from './ChooseForm';
import {ChooseEntry} from "./ChooseEntry";
import {ChooseEntryField} from "./ChooseEntryField";
import Enzyme, {shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

describe( 'Choosing controls', () => {

	const _forms = require( '../../__MOCK_DATA__/forms' )
	const forms = Object.values(_forms );
	const formId = 'CF5bde22b646ed0';
	const form = _forms[formId];
	const entries = require( '../../__MOCK_DATA__/entries' );
	const entry = entries[0];
	const entryFieldId = 'fld_8768091';
	const entryId = entry.id;
	const instanceId = '54fds';
	const beforeText = 'Before';
	const afterText = 'After';
	const onSetForm = jest.fn();
	const onSetEntry = () => jest.fn();
	const onSetEntryField =jest.fn();
	const setBefore = () => jest.fn();
	const setAfter = () => jest.fn();
	test( 'Choose form', () => {
		expect(
			renderer.create(
				<ChooseForm
					forms={forms}
					currentFormId={formId}
					onSetForm={onSetForm}
					instanceId={instanceId}
				/>
			)
		).toMatchSnapshot();
	});

	test( 'Choose form with no forms selected', () => {
		expect(
			renderer.create(
				<ChooseForm
					forms={forms}
					currentFormId={''}
					onSetForm={onSetForm}
					instanceId={instanceId}
				/>
			)
		).toMatchSnapshot();
	});

	test( 'Choose form when there no forms to choose from', () => {
		expect(
			renderer.create(
				<ChooseForm
					forms={[]}
					currentFormId={''}
					onSetForm={onSetForm}
					instanceId={instanceId}
				/>
			)
		).toMatchSnapshot();
	});


	it('ChooseForm call onSetForm prop and return value, not event', () => {
		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<ChooseForm
				forms={forms}
				currentFormId={''}
				onSetForm={mockFn}
				instanceId={instanceId}
			/>);
		component.find('select').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');
	});


	test( 'Choose entry', () => {
		expect(
			renderer.create(
				<ChooseEntry
					entries={entries}
					currentEntry={entryId}
					onSetEntry={onSetEntry}
					instanceId={instanceId}
				/>
			)
		).toMatchSnapshot();
	});


	it('ChooseEntry call onSetEntry and return value, not event', () => {
		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<ChooseEntry
				entries={entries}
				currentEntry={entryId}
				onSetEntry={mockFn}
				instanceId={instanceId}
			/>);
		component.find('select').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');
	});

	test( 'Choose entry field', () => {
		expect(
			renderer.create(
				<ChooseEntryField
					currentEntry={entryId}
					form={form}
					onSetField={onSetEntryField}
					instanceId={instanceId}
					entries={entries}
					entryFieldId={entryFieldId}
				/>
			)
		).toMatchSnapshot();
	});

	it('ChooseEntryField call onSetField and return value, not event', () => {
		const mockFn = jest.fn();
		const event = {
			preventDefault() {},
			target: { value: 'matched' }
		};
		const component = mount(
			<ChooseEntryField
				currentEntry={entryId}
				form={form}
				onSetField={mockFn}
				instanceId={instanceId}
				entries={entries}
				entryFieldId={entryFieldId}
			/>);
		component.find('select').simulate('change', event);
		expect(mockFn.mock.calls[0][0]).toBe('matched');
	});

});
