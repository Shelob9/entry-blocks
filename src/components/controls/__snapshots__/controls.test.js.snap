exports[`Choosing controls Choose entry 1`] = `
Array [
  <label
    htmlFor="caldera-forms-entry-chooser-54fds">
    Choose Entry
</label>,
  <select
    className="caldera-forms-entry-chooser"
    id="caldera-forms-entry-chooser-54fds"
    onChange={[Function]}
    value="1">
    <option
        value={null}>
        --
    </option>
    <option
        value="1">
        1
    </option>
    <option
        value="2">
        2
    </option>
</select>,
]
`;

exports[`Choosing controls Choose entry field 1`] = `
Array [
  <label
    className=""
    htmlFor="caldera-forms-entry-field-chooser-54fds">
    Choose Field
</label>,
  <select
    className="caldera-forms-entry-field-chooser"
    id="caldera-forms-entry-field-chooser-54fds"
    onChange={[Function]}
    value="fld_8768091">
    <option
        value={null}>
        --
    </option>
    <option
        value="id">
        ID
    </option>
    <option
        value="datestamp">
        Submitted
    </option>
    <option
        value="fld_8768091">
        First Name
    </option>
    <option
        value="fld_9970286">
        Last Name
    </option>
    <option
        value="fld_6009157">
        Email Address
    </option>
    <option
        value="fld_7683514">
        Comments / Questions
    </option>
</select>,
]
`;

exports[`Choosing controls Choose form 1`] = `
<div>
  <label
    className=""
    htmlFor="caldera-forms-form-chooser-54fds">
    Choose Form
  </label>
  <select
    className="caldera-forms-form-chooser"
    id="caldera-forms-form-chooser-54fds"
    onChange={[Function]}
    value="CF5bde22b646ed0">
    <option>
       -- 
    </option>
    <option
      value="CF5bde22a843265">
      Not Contact Form
    </option>
    <option
      value="CF5bde22b646ed0">
      Contact Form
    </option>
  </select>
</div>
`;

exports[`Choosing controls Choose form when there no forms to choose from 1`] = `
<div>
  <label
    className=""
    htmlFor="caldera-forms-form-chooser-54fds">
    Choose Form
  </label>
  <select
    className="caldera-forms-form-chooser"
    id="caldera-forms-form-chooser-54fds"
    onChange={[Function]}
    value="">
    <option>
       -- 
    </option>
  </select>
</div>
`;

exports[`Choosing controls Choose form with no forms selected 1`] = `
<div>
  <label
    className=""
    htmlFor="caldera-forms-form-chooser-54fds">
    Choose Form
  </label>
  <select
    className="caldera-forms-form-chooser"
    id="caldera-forms-form-chooser-54fds"
    onChange={[Function]}
    value="">
    <option>
       -- 
    </option>
    <option
      value="CF5bde22a843265">
      Not Contact Form
    </option>
    <option
      value="CF5bde22b646ed0">
      Contact Form
    </option>
  </select>
</div>
`;
